packages/mgt/src/utils/Cache.ts

```diff
@@ -0,0 +1,361 @@
+}
+
+/**
+ * class in charge of managing all the caches and their stores
+ *
+ * @export
+ * @class CacheService
+ */
+// tslint:disable-next-line: max-classes-per-file
+export class CacheService {
+  /**
+   *  Looks for existing cache, otherwise creates a new one
+   *
+   * @static
+   * @template T
+   * @param {CacheSchema} schema
+   * @param {string} store
+   * @returns {Cache<T>}
+   * @memberof CacheService
+   */
+  public static getCache<T extends CacheItem>(schema: CacheSchema, store: string): Cache<T> {
+    const key = `${schema.name}/${store}`;
+
+    if (!this.isInitialized) {
+      this.init();
+    }
+
+    if (!this.cacheStore.has(store)) {
+      this.cacheStore.set(key, new Cache<T>(schema, store));
+    }
+    return this.cacheStore.get(key) as Cache<T>;
+  }
+
+  /**
+   * Globally enables the cache
+   *
+   * @static
+   * @memberof CacheService
+   */
+  public static enableCacheGlobal() {
+    CacheService._cacheConfig = {
+      defaultInvalidationPeriod: 3600000,
+      people: {
+        invalidiationPeriod: CacheService.config.people.invalidiationPeriod,
+        isEnabled: true
+      },
+      photos: {
+        invalidiationPeriod: CacheService.config.photos.invalidiationPeriod,
+        isEnabled: true
+      },
+      users: {
+        invalidiationPeriod: CacheService.config.users.invalidiationPeriod,
+        isEnabled: true
+      }
+    };
+  }
+
+  /**
+   * Globally disables the cache
+   *
+   * @static
+   * @memberof CacheService
+   */
+  public static disableCacheGlobal() {
+    CacheService._cacheConfig = {
+      defaultInvalidationPeriod: 3600000,
+      people: {
+        invalidiationPeriod: CacheService.config.people.invalidiationPeriod,
+        isEnabled: false
+      },
+      photos: {
+        invalidiationPeriod: CacheService.config.photos.invalidiationPeriod,
+        isEnabled: false
+      },
+      users: {
+        invalidiationPeriod: CacheService.config.users.invalidiationPeriod,
+        isEnabled: false
+      }
+    };
+  }
+
+  private static cacheStore: Map<string, Cache<CacheItem>> = new Map();
+  private static isInitialized: boolean = false;
+  private static globalEnabled: boolean = true;
+  private static _state: ProviderState;
+
+  private static _cacheConfig: CacheConfig = {
+    defaultInvalidationPeriod: 3600000,
+    people: {
+      invalidiationPeriod: null,
+      isEnabled: true
+    },
+    photos: {
+      invalidiationPeriod: null,
+      isEnabled: true
+    },
+    users: {
+      invalidiationPeriod: null,
+      isEnabled: true
+    }
+  };
+
+  /**
+   * returns the cacheconfig object
+   *
+   * @readonly
+   * @static
+   * @type {CacheConfig}
+   * @memberof CacheService
+   */
+  public static get config(): CacheConfig {
+    return this._cacheConfig;
+  }
+
+  /**
+   * @private
+   * @static
+   * @memberof CacheService
+   */
+  private static init() {
+    if (Providers.globalProvider) {
+      this._state = Providers.globalProvider.state;
```

Comment:
Since `this._state` is only used in this block, you can define the variable here instead of defining it as a static field. Also, consider renaming it to `previousState` for readability